### PR TITLE
Added property clusters

### LINKED ISSUE #341: Add property clusters to aa.py
For each amino acid property, manually cluster amino acids based on that property. Add the clusters of amino acids to aa.py

### PR DESCRIPTION
**Pull request by bamueh**
Fixes #341

### CODE REVIEW

File: dark/aa.py [hunk 1]
```diff
@@ -409,6 +411,257 @@
...
+PROPERTY_CLUSTERS = {
+    'A': {
+        'aliphaticity': 1,
+        'aromaticity': 1,
+        'composition': 1,
+        'hydrogenation': 1,
+        'hydropathy': 3,
+        'hydroxyethilation': 2,
+        'iep': 2,
+        'polar_req': 2,
```
Comment (terrycojones) [inline dark/aa.py]: Why the underscore? What's `polar_req` an abbreviation for anyway?

**Comment by terrycojones** [comment]
I can think of several tests you could do on that dict.

File: dark/aa.py (hunk 1 above)
Comment (bamueh) [inline dark/aa.py]: It's the abbreviation for 'polar requirement'. I can change it to `polar_requirement`.

**Comment by terrycojones** [comment]
Here are some tests I'd add. These are all to check against accidental mistakes in that dict (either present now or introduced later in an editing accident).

* The set of keys in the dict must match `dark.aa.AA_LETTERS`.
* Each sub-dict must have exactly the set of expected keys.
* For each property, the set of values must be 1, 2, 3... with no omissions.
* (Covered by the above) There must not be a zero value for any property of any AA.

File: test/test_aa.py [hunk 2]
```diff
@@ -390,3 +462,43 @@ def testFindByNameCaseIgnoredNameWithSpace(self):
...
+        """
+        self.assertEqual(AA_LETTERS, sorted(PROPERTY_CLUSTERS.keys()))
+
+    def testNumberOfValues(self):
+        """
+        Each key in PROPERTY_CLUSTERS must have a dict with 10 elements in it
+        as the value.
+        """
+        for aa, propertiesDict in PROPERTY_CLUSTERS.items():
+            self.assertIs(10, len(propertiesDict))
```
Comment (terrycojones) [inline test/test_aa.py]: This is a lazy test :-)  Check the actual values, not how many there are.

File: test/test_aa.py [hunk 3]
```diff
@@ -390,3 +462,43 @@ def testFindByNameCaseIgnoredNameWithSpace(self):
...
+        The PROPERTY_CLUSTERS dict must contain the right keys.
+        """
+        self.assertEqual(AA_LETTERS, sorted(PROPERTY_CLUSTERS.keys()))
+
+    def testNumberOfValues(self):
+        """
+        Each key in PROPERTY_CLUSTERS must have a dict with 10 elements in it
+        as the value.
+        """
+        for aa, propertiesDict in PROPERTY_CLUSTERS.items():
```
Comment (terrycojones) [inline test/test_aa.py]: Here I think you want `PROPERTY_CLUSTERS.values()` seeing as you don't use `aa` in the loop.

File: test/test_aa.py [hunk 4]
```diff
@@ -390,3 +462,43 @@ def testFindByNameCaseIgnoredNameWithSpace(self):
...
+            self.assertIs(10, len(propertiesDict))
+
+    def testAliphaticity(self):
+        """
+        Aliphaticity must always be in cluster 1.
+        """
+        for aa, propertiesDict in PROPERTY_CLUSTERS.items():
```
Comment (terrycojones) [inline test/test_aa.py]: `.values()` again.

**Pull request merged**